sparse/src/KokkosSparse_IOUtils.hpp

```diff
@@ -73,7 +73,8 @@ void kk_sparseMatrix_generate(OrdinalType nrows, OrdinalType ncols, SizeType &nn
   }
   // Sample each value from uniform (-50, 50) for real types, or (-50 - 50i, 50
   // + 50i) for complex types.
-  Kokkos::View<ScalarType *, Kokkos::HostSpace> valuesView(values, nnz * block_elem_count);
+  Kokkos::View<ScalarType *, Kokkos::HostSpace> valuesView(
+      values, nnz * static_cast<typename Kokkos::View<ScalarType *, Kokkos::HostSpace>::size_type>(block_elem_count));
```

Comment:
```suggestion
  Kokkos::View<ScalarType *, Kokkos::HostSpace> valuesView(
      values, size_t(nnz) * block_elem_count);
```
?

Reply:
Here we could actually try to use the `SizeType` template parameter but that could lead to issues if the user picks something like `int` for that type? We could also define an `impl_size_type` that is not visible to the user and picks the right type?